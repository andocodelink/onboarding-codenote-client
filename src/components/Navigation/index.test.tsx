import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount, render } from 'enzyme';
import { Button } from 'react-bootstrap';

import { Navigation, mapStateToProps, mapDispatchToProps } from './index';

describe('Home screen', () => {
  const userHasAuthenticated = () => ({});
  const jdomAlert = window.alert;

  beforeAll(() => {
    window.alert = jest.fn();
  });

  afterAll(() => {
    window.alert = jdomAlert;
  })
  
  test('renders', () => {
    let history: any = jest.fn();
    const wrapper = render(<MemoryRouter><Navigation isAuthenticated={true} history={history} userHasAuthenticated={(() => ({}))} /></MemoryRouter>);
    expect(wrapper).toMatchSnapshot();
  });

  test('logout handler', async () => {
    let history: any = jest.fn();
    history.push = (..._) => ({})
    let expectedIsAuthen = true;
    let app = new Navigation({
      isAuthenticated: true,
      history: history,
      userHasAuthenticated: ((isAuthen) => {
        expectedIsAuthen = isAuthen;
        return { expectedIsAuthen };
      })
    });
    await app.handleLogout();
    expect(expectedIsAuthen).toBe(false);
  });

  test('map state to props', () => {
    let props = mapStateToProps({
      authenticate: {
        isAuthenticated: true 
      }
    });
    expect(props.isAuthenticated).toBe(true);
  });
  test('map dispatch to props', () => {
    let dispatch = jest.fn();
    let props =  mapDispatchToProps(dispatch);
    expect(props.userHasAuthenticated).toBeInstanceOf(Function);
  });

})