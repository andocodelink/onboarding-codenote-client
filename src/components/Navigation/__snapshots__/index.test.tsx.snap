// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen renders 1`] = `
<nav
  class="navbar navbar-default"
>
  <div
    class="container-fluid"
  >
    <div
      class="navbar-header"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        CodeNote
      </a>
      <button
        class="navbar-toggle collapsed"
        type="button"
      >
        <span
          class="sr-only"
        >
          Toggle navigation
        </span>
        <span
          class="icon-bar"
        />
        <span
          class="icon-bar"
        />
        <span
          class="icon-bar"
        />
      </button>
    </div>
    <div
      class="navbar-collapse collapse"
    >
      <ul
        class="nav navbar-nav navbar-right"
      >
        <li
          class=""
          role="presentation"
        >
          <a
            href="#"
            role="button"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
